runtimes/core/src/api/manager.rs

```diff
@@ -375,6 +375,10 @@ impl Manager {
                 res = async { api_fut.unwrap().await }, if api_fut.is_some() => {
                     res.context("serve api").inspect_err(|err| log::error!("api server failed: {:?}", err))?;
                 },
+                else => {
+                    // Nothing to serve.
+                    ::log::debug!("no api server or gateway to serve");
```

Comment:
Maybe this should be a warn or error? Assuming that something would be wrong if we have nothing to serve?

Reply:
It's actually expected in some situations that I encountered yesterday, so that's why I made it a debug statement